pkg/controller/mizar/mizar-arktos-network-controller.go

```diff
@@ -172,221 +168,192 @@ func (c *MizarArktosNetworkController) syncNetwork(eventKeyWithType KeyWithEvent
+	klog.V(3).Infof("Created VPC (%s) and Subnet(%s) for tenant %s successfully", vpc, subnet, network.Tenant)
 
 	return nil
 }
 
-func createVpcAndSubnetCRD(tenant, vpc, subnet, vpcDefaultTemplatePath, subnetDefaultTemplatePath string, discoveryClient discovery.DiscoveryInterface, dynamicClient dynamic.Interface) error {
-	// Create VPC CRD
-	err := createVpcOrSubnetCRD(tenant, vpc, vpcDefaultTemplatePath, discoveryClient, dynamicClient)
+func createVpcAndSubnet(vpc, subnet string, discoveryClient discovery.DiscoveryInterface, dynamicClient dynamic.Interface) error {
+	// Create VPC object
+	vpcIpStart, vpcSpec := generateVPCSpec(vpc)
+	vpcData, err := json.Marshal(vpcSpec)
 	if err != nil {
-		klog.Errorf("Create actual VPC object (%s) in error (%v).", vpc, err)
+		klog.Errorf("Error marshalling VPC %s spec. Error: %v", vpc, err)
 		return err
 	}
-	klog.V(4).Infof("Complete to create VPC: (%s) successfully", vpc)
-
-	// Create Subnet CRD
-	err = createVpcOrSubnetCRD(tenant, subnet, subnetDefaultTemplatePath, discoveryClient, dynamicClient)
-	if err != nil {
-		klog.Errorf("Create actual Subnet object (%s) in error (%v).", subnet, err)
+	err = createUnstructuredObject(vpcData, discoveryClient, dynamicClient)
+	if err != nil && !apierrors.IsAlreadyExists(err) {
+		klog.Errorf("Error creating VPC %s. Error: %v", vpc, err)
 		return err
 	}
-	klog.V(4).Infof("Complete to create VPC: (%s) and Subnet: (%s) successfully", vpc, subnet)
-
-	return nil
-}
 
-func createVpcOrSubnetCRD(tenant, vpcOrSubnetName, defaultTemplatePath string, discoveryClient discovery.DiscoveryInterface, dynamicClient dynamic.Interface) error {
-	// Get json file in bytes format
-	manifestData, err := GetCRDVpcOrSubnetSpec(defaultTemplatePath, vpcOrSubnetName, tenant)
+	// Create Subnet object
+	subnetSpec, err := generateSubnetSpec(vpc, subnet, vpcIpStart)
 	if err != nil {
-		klog.Errorf("At (%s) get JSON spec for CRD (%s) in error: (%v)", defaultTemplatePath, vpcOrSubnetName, err)
+		klog.Errorf("Error getting Subnet %s spec. Error: %v", subnet, err)
 		return err
 	}
-
-	// Create VPC or Subnet object
-	err = createUnstructuredObject([]byte(manifestData), vpcOrSubnetName, discoveryClient, dynamicClient)
-	if err != nil {
-		klog.Errorf("Create CRD object (%s) in error: (%v)", vpcOrSubnetName, err)
+	err = createUnstructuredObject(subnetSpec, discoveryClient, dynamicClient)
+	if err != nil && !apierrors.IsAlreadyExists(err) {
+		klog.Errorf("Error creating Subnet %s for VPC %s. Error: %v", subnetSpec, vpc, err)
 		return err
 	}
-	klog.V(3).Infof("Create CRD object: (%s) successfully", vpcOrSubnetName)
 
 	return nil
 }
 
-func createUnstructuredObject(data []byte, vpcOrSubnetName string, discoveryClient discovery.DiscoveryInterface, dynamicClient dynamic.Interface) error {
+func createUnstructuredObject(data []byte, discoveryClient discovery.DiscoveryInterface, dynamicClient dynamic.Interface) error {
```

Comment:
This function seems can use further refactoring. Leave it for other PR for scope limit.